format/macho/testdata/darwin_amd64/libbbb.a.fqtest

```diff
@@ -0,0 +1,31 @@
+     |00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f|0123456789abcdef|.{}: darwin_amd64/libbbb.a (ar) 0x0-0x3b7.7 (952)
```

Comment:
Any idea what format is used in archive on macOS? On linux the .o files inside archives turned out to be ELF, i for some reason assumed it used something else

Reply:
So the file can be a plain macho file (as we have been parsing until now) a fat file or an ar file.

Currently what I am implement on my local

Check if it is plain
If not check if it is FAT,
If not try AR file parsing,
And if that also fails say that this file has an invalid magic identifier.